Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
+        // already promisified
+        Document : processDocument,
+        // already promisified
+        Folder : processFolder,
+        Placemark : deferAndWrapInPromise(processPlacemark),
+        NetworkLink : deferAndWrapInPromise(processNetworkLink),
+        GroundOverlay : deferAndWrapInPromise(processGroundOverlay),
+        PhotoOverlay : deferAndWrapInPromise(processUnsupported),
+        ScreenOverlay : deferAndWrapInPromise(processUnsupported),
+        Tour : deferAndWrapInPromise(processUnsupported)
+    };
+
+    function deferAndWrapInPromise(originalFunction){
+        return function(){
+            var deferred = when.defer();
+            var args = Array.prototype.slice.call(arguments);
+            var context = this;
+            setTimeout(function deferredFunc(){
```

Comment:
I've been doing more research and really diving into event loops and microtasks.  There definitely are some issues with `setTimeout` and I'm sure you've seen that it does take longer to process now.  Luckily there are some clever ways out there that let you schedule a task on the next tick of the event loop without using setTimeout.  

Resources:  
https://www.nczonline.net/blog/2013/07/09/the-case-for-setimmediate/
And a polyfill that we could use to replace the `setTimeout` call:
https://github.com/YuzuJS/setImmediate

The demo / test site is very compelling: 
http://jphpsf.github.io/setImmediate-shim-demo/

As another aside, opening devtools on this github page, and typing 'setImmediate` I see that github is using what appears to be this exact polyfill (though it's minified).

Thoughts?